apps/wallet/src/ui/app/hooks/useGetTxnRecipientAddress.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useMemo } from 'react';
+
+import { getEventsSummary, getAmount } from '_helpers';
+
+import type { SuiTransactionResponse, SuiAddress } from '@mysten/sui.js';
+
+type Props = {
+    txn: SuiTransactionResponse;
+    address: SuiAddress;
+};
+
+export function useGetTxnRecipientAddress({ txn, address }: Props) {
+    const { certificate } = txn;
+    const { coins: eventsSummary } = getEventsSummary(txn.effects, address);
```

Comment:
I think we should memo this one as well, otherwise  eventsSummary will always be a new array